keps/0001a-meta-kep-implementation.md

```diff
@@ -0,0 +1,217 @@
+|   |   |   ├── experience_reports (required) # user feedback
+|   |   |   │   ├── alpha-feedback.md
+|   |   |   │   └── beta-feedback.md
+|   |   |   ├── features # units of work that span approximately one release cycle
+|   |   |   │   ├── feature-01.md
+|   |   |   │   ├── ...
+|   |   |   │   └── feature-n.md
+|   |   |   ├── guides
+|   |   |   |   ├── guide-for-developers.md (required)
+|   |   |   |   ├── guide-for-teachers.md (required)
+|   |   |   |   ├── guide-for-operators.md (required)
+|   |   |   |   └── guide-for-project-maintainers.md
+|   |   |   ├── index.json (required) # used for site generation e.g., Hugo
+|   |   |   ├── metadata.yaml (required) # used for automation / project tracking
+|   |   └── └── OWNERS
+│   ├── sig-foo
+|   ├── ...
+|   └── sig-n
+```
+
+metadata.yaml would contain information that was previously in a KEP's YAML front-matter:
+
+```
+---
+authors: # required
+  - "calebamiles" # just a GitHub handle for now
+  - "jbeda"
+title: "Kubernetes Enhancement Proposal process"
+number: 42 # required
+owning-sig: "sig-pm" # required
+participating-sigs:
+  - "sig-architecture"
+  - "sig-contributor-experience"
+approvers: # required
+  - "bgrant0607" # just a GitHub handle for now
+reviewers:
+  - "justaugustus"  # just a GitHub handle for now
+  - "jdumars"
+editors:
+  - null # generally omit empty/null fields
+status: "active" # required
+github:
+  issues:
+    - null # GitHub url
+  pull_requests:
+    - null # GitHub url
+  projects:
+    - project_id: null
+      card_id: null
+releases: # required
+  - k8s_version: v1.9
+    kep_status: "active"
+    k8s_status: "alpha" # one of alpha|beta|GA
+  - k8s_version: v1.10
+    kep_status: "active"
+    k8s_status: "alpha"
+replaces:
+  - kep_location: null
+superseded-by:
+  - kep_location: null
+created: 2018-01-22 # in YYYY-MM-DD
+updated: 2018-09-04
+```
+
+- Move existing KEPs into [k/features]
+- Create a `kind/kep` label for [k/community] and [k/features]
+  - For `k/community`:
+    - Label incoming KEPs as `kind/kep`
+    - Enable searches of `org:kubernetes label:kind/kep`, so we can identify active PRs to `k/community` and reroute the PR authors to `k/enhancements` (depending on the state)
+  - For `k/enhancements` (fka `k/features`):
+    - Label incoming KEPs as `kind/kep`
+    - Classify KEP submissions / tracking issues as `kind/kep`, differentiating them from `kind/feature`
+- Move existing design proposals into [k/features]
+- Move existing architectural documents into [k/features] (process TBD)
+- Deprecate design proposals
+- Rename [k/features] to [k/enhancements]
+- Create tombstones / redirects to [k/enhancements]
+- Prevent new KEPs and design proposals from landing in [k/community]
+- Remove `kind/kep` from [k/community] once KEP migration is complete
+- Correlate existing Feature tracking issues with links to KEPs
+- Fix [KEP numbering races] by using the GitHub issue number of the KEP tracking issue
```

Comment:
I'm not a huge fan of this but I guess it is ok.  We'll have sparse numbering as there will be other PRs.

One thing that I want to do regardless is encourage people to check in early.  If the latency between PR for new kep and commit is small these races will be minimal.  We can and should build some presubmit to prevent races too.

In any case, if we use the PR number as the KEP number there will still have to be a dance where people will create a change, submit a PR, update their files and submit/push another change to that PR.  That dance isn't obvious and will make this a bit harder.